src/allmydata/test/web/test_web.py

```diff
@@ -1035,6 +1035,122 @@ def _check_retrieve(res):
+
+    def test_status_down_subpath(self):
+        # See that "GET /status/down-0" works.
+        h = self.s.get_history()
+        dl_num = h.list_all_download_statuses()[0].get_counter()
+        d = self.GET("/status/down-{}".format(dl_num), return_response=True)
+        def _download_status_check(result):
+            body, status, _ = result
+            self.failUnlessReallyEqual(int(status), 200)
+            soup = BeautifulSoup(body, 'html5lib')
+            assert_soup_has_favicon(self, soup)
+            assert_soup_has_tag_with_content(
+                self, soup, u"title",
+                u"Tahoe-LAFS - File Download Status"
+            )
+        d.addCallback(_download_status_check)
+        return d
+
+    def test_status_mapupdate_subpath(self):
+        # See that "GET /status/mapupdate-0" works.
+        h = self.s.get_history()
+        mu_num = h.list_all_mapupdate_statuses()[0].get_counter()
+        d = self.GET("/status/mapupdate-{}".format(mu_num), return_response=True)
+        def _mapupdate_status_check(result):
+            body, status, _ = result
+            self.failUnlessReallyEqual(int(status), 200)
+            soup = BeautifulSoup(body, 'html5lib')
+            assert_soup_has_favicon(self, soup)
+            assert_soup_has_tag_with_content(
+                self, soup, u"title",
+                u"Tahoe-LAFS - Mutable File Servermap Update Status"
+            )
+        d.addCallback(_mapupdate_status_check)
+        return d
+
+    def test_status_publish_subpath(self):
+        # See that "GET /status/publish-0" works.
+        h = self.s.get_history()
+        pub_num = h.list_all_publish_statuses()[0].get_counter()
+        d = self.GET("/status/publish-{}".format(pub_num), return_response=True)
+        def _publish_status_check(result):
+            body, status, _ = result
+            self.failUnlessReallyEqual(int(status), 200)
+            soup = BeautifulSoup(body, 'html5lib')
+            assert_soup_has_favicon(self, soup)
+            assert_soup_has_tag_with_content(
+                self, soup, u"title",
+                u"Tahoe-LAFS - Mutable File Publish Status"
+            )
+        d.addCallback(_publish_status_check)
+        return d
+
+    def test_status_retrieve_subpath(self):
+        # See that "GET /status/retrieve-0" works.
+        h = self.s.get_history()
+        ret_num = h.list_all_retrieve_statuses()[0].get_counter()
+        d = self.GET("/status/retrieve-{}".format(ret_num), return_response=True)
+        def _retrieve_status_check(result):
+            body, status, _ = result
+            self.failUnlessReallyEqual(int(status), 200)
+            soup = BeautifulSoup(body, 'html5lib')
+            assert_soup_has_favicon(self, soup)
+            assert_soup_has_tag_with_content(
+                self, soup, u"title",
+                u"Tahoe-LAFS - Mutable File Retrieve Status"
+            )
+        d.addCallback(_retrieve_status_check)
+        return d
```

Comment:
Thanks for adding the new test coverage.  All of these new `test_status_...` tests look pretty similar.  I expect most of the code could be factored into a shared helper to avoid the repetition?

Reply:
Good idea, that should have been obvious in the first place.  I grew lazy at this point. :-)

Added a helper in 95642ae.